packages/nimble-components/src/table/template.ts

```diff
@@ -47,14 +50,25 @@ export const template = html<Table>`
                             </${checkboxTag}>
                         </span>
                     `)}
-
+                    <span class="collapse-button-container" role="gridcell">
+                        <${buttonTag}
+                            class="collapse-all-button"
+                            content-hidden
+                            appearance="${ButtonAppearance.ghost}"
+                            @click="${x => x.handleCollapseAllGroupRows()}"
+                            style="--ni-private-collapse-button-visibility: ${x => (x.showCollapseAll ? 'visible' : 'hidden')}"
```

Comment:
For consistency with the rest of our repo, I think this should use the `when` directive rather than providing a hidden styling.

Reply:
I was using the CSS `visibility` property because that is how I maintain the reserved space. The `when` will remove the div.

Reply:
Ah, that makes sense...it's 'hidden' rather than 'collapsed'. In that case, ignore the comment about using `when`.

However...
I'm not sure if we have any explicit nimble guidelines about styling with CSS classes vs CSS variables set in the template. But, my leaning is that we should use CSS classes whenever possible/practical and only use variables set in templates in more complex cases. Examples of those more complex cases are when the value is not a one-to-one mapping to a style (e.g. indention level that is used in a CSS calculation) and when you need to bubble some styling state to child components. Since neither is the case here, I'd vote for having a CSS class get applied based on `showCollapseAll` and then put the styling the `styles` file. That keeps the style mostly consolidated in a single place.

Reply:
I'm unresolving this conversation. I think it was marked as resolved before my most recent comment was made.

Reply:
Done.